store/cachemulti/store.go

```diff
@@ -167,3 +167,38 @@ func (cms Store) GetKVStore(key types.StoreKey) types.KVStore {
 	}
 	return store.(types.KVStore)
 }
+
+//----------------------------------------
+// fork only
+
+// Clone a method a added to deep clone the state of the Store
+func (cms Store) Clone() types.CacheMultiStore {
+
+	// her based on the following documentation from CacheWrap:
+	// ---
+	// CacheWrap is the most appropriate interface for store ephemeral branching and cache.
+	// For example, IAVLStore.CacheWrap() returns a CacheKVStore. CacheWrap should not return
+	// a Committer, since Commit ephemeral store make no sense. It can return KVStore,
+	// HeapStore, SpaceStore, etc.
+	// ---
+	// We can assument that any of the types behind CacheWrap will implement types.CacheKVStore
+	// because of this, we implement the Clone Method on the CacheKVStore
+	// Which implement both Store and CacheWrapper
+	stores := make(map[types.StoreKey]types.CacheWrap, len(cms.stores))
+	for k, v := range cms.stores {
+		stores[k.Clone()] = v.(types.CacheKVStore).Clone()
```

Comment:
This type conversion makes me itchy 😅 I see we don't have other choice but just in case, can we check the conversion result and panic in case of failure with a clear message? At least we will ensure visibility in case of any problem her, for example, after a breaking Cosmos upgrade that will use a different implementation of `types.CacheWrap`.

Reply:
sure.

Reply:
addressed in: b60c017209726a39b2e133dec9abef26e53b0185